dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -106,9 +134,29 @@ public TrackedEntityQueryParams map(TrackedEntityOperationParams operationParams
         .setIncludeDeleted(operationParams.isIncludeDeleted())
         .setPotentialDuplicate(operationParams.getPotentialDuplicate());
 
+    validateGlobalSearchParameters(params);
+
     return params;
   }
 
+  private List<TrackedEntityType> getTrackedEntityTypes(
+      TrackedEntityType trackedEntityType, Program program, User user) {
+
+    if (program != null && program.getTrackedEntityType() != null) {
+      return List.of(program.getTrackedEntityType());
+    } else if (trackedEntityType == null) {
+      return filterAndValidateTrackedEntityTypes(user);
+    }
+
+    return emptyList();
+  }
+
+  private List<TrackedEntityType> filterAndValidateTrackedEntityTypes(User user) {
```

Comment:
I still think we should validate if user has no access to any Tracked Entity Types because if we skip this validation in the case the client specifies something in trackedEntities then we could return some results even if the user has no access to.

Reply:
When we run the query against the DB we join the tracked entity type table and we filter by the value we found here. In case there's no value, the result of the query will be an empty list, so in theory we should not leak any information. But I agree that it's better to stop the request here instead of running a query against a database when we already know we'll return an empty list.